src/utilities/geometry/Intersection.cpp

```diff
@@ -102,36 +102,113 @@ inline bool has_self_intersections(Geometry const& geometry, bool throw_on_self_
   return boost::geometry::detail::overlay::has_self_intersections(geometry, strategy, robust_policy, throw_on_self_intersection);
 }
 
+// Some operators for Boost data types that are useful for extra logging
+std::ostream& operator<<(std::ostream& os, const BoostRing& boostRing) {
+  os << "[";
+  for (unsigned int i = 0; i < boostRing.size(); i++) {
+    os << "{";
+    os << R"("X":)" << boostRing[i].x() << ",";
+    os << R"("Y":)" << boostRing[i].y() << ",";
+    os << R"("Z":0.0},)";
+  }
+  os << "],";
+  return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const std::vector<BoostRing>& boostRings) {
+  os << "[";
+  for (const auto& ring : boostRings) {
+    os << ring;
+  }
+  os << "]";
+  return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const BoostPolygon& boostPolygon) {
+  os << "{";
+  os << R"("Color":"Black",)";
+  os << R"("OuterPath":)" << boostPolygon.outer();
+  os << R"("InnerPaths":)" << boostPolygon.inners();
+  os << "},";
+  return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const std::vector<BoostPolygon>& boostPolygons) {
+  os << "[";
+  for (const auto& poly : boostPolygons) {
+    os << poly;
+  }
+  os << "]";
+  return os;
+}
+
+std::ostream& operator<<(std::ostream& os, const BoostMultiPolygon& boostPolygons) {
+  os << "[";
+  for (const auto& poly : boostPolygons) {
+    os << poly;
+  }
+  os << "]";
+  return os;
+}
+
+// Scale factor for parameters passed to boost
+double scaleBy = 1000.0;
```

Comment:
Never use a global variable like this.

```suggestion
// Scale factor for parameters when converting between openstudio and boost data formats to improve the numerical accuracy of the boolean operations.
// Idea came from this comment in the boostorg/geometry repo: https://github.com/boostorg/geometry/issues/1034#issuecomment-1284180101
// where the author indicates that scaling the values by 10 improved the result and talks about rounding to an integer grid, 
// so by increasing the range of values we pass to boost we should be improving the resolution of the integer rounding
static constexpr double scaleBy = 1000.0;
```